Marlin/src/module/stepper.cpp

```diff
@@ -2070,7 +2070,8 @@ uint32_t Stepper::block_phase_isr() {
 
       // For non-inline cutter, grossly apply power
       #if ENABLED(LASER_FEATURE) && DISABLED(LASER_POWER_INLINE)
-        cutter.apply_power(current_block->cutter_power);
+        if (cutter.enabled())
+          cutter.ocr_set_power(current_block->cutter_ocr_power);
```

Comment:
Consider the case where only a PWM pin exists, where setting the power of the laser is the way that the laser is enabled, and where the power is only ever set by the Stepper ISR because `M3`/`M4` commands don't set the power directly, but only take effect when the next planner block is processed.

Reply:
> Consider the case where only a PWM pin exists

This is not possible according to the statement from https://github.com/MarlinFirmware/Marlin/pull/22721#issuecomment-917715058

> Currently, the ENA pin is always assumed to exist,...

Let's consider how you can turn on / off the laser:

- 1) M3 / M4 S <PWM>; M5 - obviously. If not corrected, the laser will turn on with any movement (G0 include), as we ask for it.
- 2) in dynamic mode - i don't know how this in G-code, but block for this below (https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/Marlin/src/module/stepper.cpp#L2238)
- 3) In G-Code, that not here

4) And may be, if I understood right - on by start G1,G2,G3 and then turns off at the end command; move without on - G0

I tried to find a description of how the laser should work:

- https://github.com/gnea/grbl (laser mode) - It is not directly stated here, but it seems like according to [1] 
- https://smoothieware.org/laser-cutter-guide (SUPPORTED G-CODES) - Here it is possible to move the laser through G0 (https://wiki.shapeoko.com/index.php/G-Code#Code_Overview | G0/G00)

Reply:
Dynamic mode does not exist in the current code, what you referenced is acceleration and deceleration compensation to create a linear power across the lasered material. This is really a continuous mode with compensation.
In my PR I added a crude Dynamic mode which bases the power output level on a given feed rate like (GRBL). There is more work to do such as create integration with the accel/decel gradient and allow powerband settings with range floors and ceilings.

When in inline mode (M3I) the new code will not allow G0 or G28 to power the laser, it will shut it down with apply_power(0) as it should.

G1/2/3/5 and can freely power the laser in inline mode.

Then there is the "Unicorn" M106/107 (FANs) can also control the laser output in a quasi out of band PWM method. (e.g. LightBurn) I have not touched that code. 

In standard mode M3/4 should be able to turn on or off anytime provided that it is enabled.

The enable issue is still to be addressed, apply_power() is the best place to for a virtual (abstracted) enable for PWM output control. we can allow cutter.power to hold the output value and then the FSM can apply it if we are enabled.
I have changed the code to only use apply_power() across all modes.

Reply:
>> Consider the case where only a PWM pin exists

> This is not possible according to the statement

So, is it not in any future plan to allow for a laser/spindle to be controllable by PWM alone? Do any such devices exist?

Reply:
I know of many solid state lasers that are purely PWM. They are common.
![SolidStatePWM](https://user-images.githubusercontent.com/10734527/133539354-9cef561e-00c6-478a-b175-226d7ffc870e.png)